Modix.Services/Starboard/StarboardService.cs

```diff
@@ -34,31 +89,41 @@ public async Task<IUserMessage> GetFromStarboard(IGuildChannel channel, IMessage
                 .GetMessagesAsync()
                 .FlattenAsync();
 
+            //We need to store entries in the db to avoid this horrible mess
             return starMessages
                 .Cast<IUserMessage>()
-                .FirstOrDefault(x => x.Content.EndsWith(message.Id.ToString()));
+                .FirstOrDefault(x => x.Embeds
+                                        .Select(y => y.Fields.Last())
+                                        .Any(y => y.Value.Contains(message.GetJumpUrl())));
         }
 
+        /// <inheritdoc />
         public async Task RemoveFromStarboard(IGuildChannel channel, IMessage message)
         {
             var messageToRemove = await GetFromStarboard(channel, message);
             await messageToRemove.DeleteAsync();
         }
 
-        public bool IsStarReaction(SocketReaction reaction)
-            => reaction.Emote.Name == GoodEmote;
+        /// <inheritdoc />
+        public bool IsStarEmote(IEmote emote)
+            => emote.Name == _emojis.Values.Last();
 
-        public int GetReactionCount(IUserMessage message, IReaction reaction)
+        /// <inheritdoc />
+        public int GetReactionCount(IUserMessage message, IEmote emote)
         {
-            if (!message.Reactions.TryGetValue(reaction.Emote, out var metadata))
+            if (!message.Reactions.TryGetValue(emote, out var metadata))
                 return 0;
-            return message.Reactions[reaction.Emote].ReactionCount;
+            return metadata.ReactionCount;
         }
 
-        public bool IsAboveReactionThreshold(IUserMessage message, IReaction reaction)
-            => GetReactionCount(message, reaction) >= 2;
+        /// <inheritdoc />
+        public bool IsAboveReactionThreshold(IUserMessage message, IEmote emote)
+            => GetReactionCount(message, emote) >= 2;
 
-        public string BuildContextUrl(IGuildChannel channel, IMessage message)
-            => string.Join("/", _baseContextUrl, channel.Guild.Id, channel.Id, message.Author.Id);
+        public string GetStarEmote(int reactionCount)
+        {
+            var valIdx = _emojis.Keys.First((val) => reactionCount >= val);
```

Comment:
Why is this variable called `valIdx`? Is there a better name that could be used?